### PR TITLE
Streams - Quick start guide

### PR DESCRIPTION
**Pull request by Danthar**

### CODE REVIEW

File: src/docs/streams/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+```
+This line will complement the source with a consumer function--in this example we simply print out the numbers to the console--and pass this little stream setup to an Actor that runs it. This activation is signaled by having �run� be part of the method name; there are other methods that run Akka Streams, and they all follow this pattern.
+
+You may wonder where the Actor gets created that runs the stream, and you are probably also asking yourself what this materializer means. In order to get this value we first need to create an Actor system:
+```csharp
+using (var system = ActorSystem.Create("system"))
+using (var materializer = system.Materializer())
+```
+There are other ways to create a materializer, e.g. from an `ActorContext` when using streams from within Actors. The `Materializer` is a factory for stream execution engines, it is the thing that makes streams run --you don�t need to worry about any of the details just now apart from that you need one for calling any of the run methods on a `Source`. The materializer is picked up implicitly if it is omitted from the `run` method call arguments, which we will do in the following.
```
Comment (Silv3rcircl3) [inline src/docs/streams/quickstart.md]: ``` The materializer is picked up implicitly if it is omitted from the `run` method call arguments, which we will do in the following.```

That's not possible for us, we always need to pass the materializer as parameter for run methods

File: src/docs/streams/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+The nice thing about Akka Streams is that the `Source` is just a description of what you want to run, and like an architect�s blueprint it can be reused, incorporated into a larger design. We may choose to transform the source of integers and write it to a file instead:
+```csharp
+  var factorials = source.Scan(new BigInteger(1), (acc, next) => acc * next);
```
Comment (Silv3rcircl3) [inline src/docs/streams/quickstart.md]: ```BigInteger``` needs a reference to ```System.Numerics```, so for simplicity I would preffer ```1L```

File: src/docs/streams/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  var result =
+      factorials
+          .Select(num => ByteString.FromString("{num}\n"))
+          .RunWith(FileIO.ToFile(new FileInfo("factorials.txt")));
```
Comment (marcpiechura) [inline src/docs/streams/quickstart.md]: Needs to be `RunWith(FileIO.ToFile(new FileInfo("factorials.txt")), materializer);`

File: src/docs/streams/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+```
+First we use the `scan` combinator to run a computation over the whole stream: starting with the number 1 `(BigInteger(1))` we multiple by each of the incoming numbers, one after the other; the `scan` operation emits the initial value and then every calculation result. This yields the series of factorial numbers which we stash away as a Source for later reuse�it is important to keep in mind that nothing is actually computed yet, this is just a description of what we want to have computed once we run the stream. Then we convert the resulting series of numbers into a stream of `ByteString` objects describing lines in a text file. This stream is then run by attaching a file as the receiver of the data. In the terminology of Akka Streams this is called a `Sink. IOResult` is a type that IO operations return in Akka Streams in order to tell you how many bytes or elements were consumed and whether the stream terminated normally or exceptionally.
```
Comment (Silv3rcircl3) [inline src/docs/streams/quickstart.md]: If you change ```BigInterger``` you would also need to change it here

File: src/docs/streams/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+```
+This line will complement the source with a consumer function--in this example we simply print out the numbers to the console--and pass this little stream setup to an Actor that runs it. This activation is signaled by having �run� be part of the method name; there are other methods that run Akka Streams, and they all follow this pattern.
+
+You may wonder where the Actor gets created that runs the stream, and you are probably also asking yourself what this materializer means. In order to get this value we first need to create an Actor system:
+```csharp
+using (var system = ActorSystem.Create("system"))
+using (var materializer = system.Materializer())
+```
+There are other ways to create a materializer, e.g. from an `ActorContext` when using streams from within Actors. The `Materializer` is a factory for stream execution engines, it is the thing that makes streams run --you don�t need to worry about any of the details just now apart from that you need one for calling any of the run methods on a `Source`. The materializer is picked up implicitly if it is omitted from the `run` method call arguments, which we will do in the following.
```
Comment (Silv3rcircl3) [inline src/docs/streams/quickstart.md]: ``` The materializer is picked up implicitly if it is omitted from the `run` method call arguments, which we will do in the following.```

That's not possible for us, we always need to pass the materializer as parameter for run methods

Comment (Silv3rcircl3) [inline src/docs/streams/quickstart.md]: ``` The materializer is picked up implicitly if it is omitted from the `run` method call arguments, which we will do in the following.```

That's not possible for us, we always need to pass the materializer as parameter for run methods

**Comment by marcpiechura** [comment]
I don't know why but I can't add any additional comments, Github always say's `commented on a outdated diff`... So I add them here... 

1.
`The materializer is picked up implicitly if it is omitted from the `run` method call arguments, which we will do in the following.`

That's not possible for us, we always need to pass the materializer as parameter for run methods  

2.
`BigInteger` needs a reference to `System.Numerics`, so for simplicity I would preffer `1L`

**Comment by Danthar** [comment]
1: True. That was one thing that i still needed to remove.
2: A long goes negative pretty fast in this sample. In order to produce a proper output on par with the JVM example a BigInteger is needed.

**Pull request merged**